hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -1451,7 +1361,7 @@ public AppActivitiesInfo getAppActivities(HttpServletRequest hsr,
       long startTime = clock.getTime();
       SubClusterInfo subClusterInfo = getHomeSubClusterInfoByAppId(appId);
```

Comment:
Create one with appId and then you do the proper check for null, etc.

Reply:
I will modify the code.